src/lib/utilities.js

```diff
@@ -238,6 +238,35 @@ export const toSigFigs = (number, precision) => {
   return String(roundedValue);
 };
 
+export const scientificNotation = (number, precision, returnHtml = true) => {
+  if (number === 0) return 0;
+
+  const isOutsideRange = number > 1 || number < -1;
+  const afterTheDecimal = String(number).split('.');
+  const isInsideRange = number > -1 && number < 1;
+  const decimalMoreThanThreeChars =
+    afterTheDecimal[1] && afterTheDecimal[1].length > 3;
+
+  if (
+    (isOutsideRange && String(number).length > 9) ||
+    (isInsideRange && decimalMoreThanThreeChars)
+  ) {
+    const value = Number.parseFloat(number).toExponential(precision || 3);
```

Comment:
This is an excellent repurposing of js functionality that already existed on the `Float` prototype.  I think we can simplify this `scientificNotation` even more by checking what we get from `toExponential` instead of counting "places", and running all these difference checks on the input number ourselves.  Something like:
```
export const scientificNotation = (
  number,
  precision = 3,
  returnHtml = true
) => {
  if (number === 0) return 0;
  const [val, exp] = Number.parseFloat(number)
    .toExponential()
    .split('e');
  const numPlaces = Math.abs(+exp);

  // Should scientificNotationize
  if ((Math.abs(number) < 1 && numPlaces > 3) || numPlaces > 9) {
    const coefficientAndBase = `${parseFloat(toSigFigs(+val, 3))} x 10`;

    return returnHtml
      ? renderDef(`${coefficientAndBase}<sup>${+exp}</sup>`)
      : `${coefficientAndBase}^${+exp}`;
  }

  // Should sigFigIt instead
  const sigFigged = addTheCommas(toSigFigs(number, precision));
  return returnHtml ? renderDef(sigFigged) : sigFigged;
};
```

Feels simpler to me at least, whatchuthink?

If you like the above then please test it however you tested your work 'cause I'm not totally confident I'm thinking of every edge case.  I think if you wanna keep your method as is there are still a couple other lil fixes:

- the precision you're handing `toExponential` is how many digits past the decimal, not how many sig figs.  Sig figs are all important digits, not just those to the right of the decimal.  I got around this in my above example by not sigFig-ifying until constructing the string/html this method returns (i.e. letting `toExponential` run with no args).  I think this is why your method isn't always trimming trailing zeroes in the string/html output.
- In my testing, `number` was sometimes a float and `String(number).length > 9` is not working when `number` is a float because it's counting the decimal and those digits to the right of it in the length.  I got around this in the above by checking the exponent returned by `toExponential` instead of checking places/length of the input number.